### ISSUE TITLE
Cell cycle score 

### ISSUE THREAD

**Issue by ateissan** [jdblischak/singleCellSeq#59]
Hello,

I am trying to use the estimator for cell cycle phase.
Can you give more information on how you estimate the cell cycle phase? I found the method super close to the CellCycleScoring function from Seurat. Is it the case? Do you base your estimation on a published method? I didn't find any information in the Scientific reports paper.

Thanks,
Aurélie

**Comment by jdblischak** [comment]
Hi @ateissan! Thanks for your interest in our method.

What method of our ours did you use to estimate the cell cycle phase? Any code you may have found in this older repository (it corresponds to our publication [Tung et al., 2017, Batch effects and the effective design of single-cell gene expression studies](https://www.nature.com/articles/srep39921)) should be superseded by the repository [fucci-seq](https://github.com/jdblischak/fucci-seq) and the R package [peco](https://github.com/jhsiao999/peco), which correspond to our latest publication, [Hsiao et al., 2019, Characterizing and inferring quantitative cell cycle phase in single-cell RNA-seq data analysis](https://www.biorxiv.org/content/10.1101/526848v2).

cc @jhsiao999

**Comment by ateissan** [comment]
I am using this code : http://jdblischak.github.io/singleCellSeq/analysis/cell-cycle-post-correction.html
Is it outdated?

**Comment by jdblischak** [comment]
That is the work of @jhsiao999 and @pytung. I'll let them have the final say on the usefulness/appropriateness of that method (they have both moved on to industry jobs, so they won't be able to respond right away).

For more details, see the companion file https://jdblischak.github.io/singleCellSeq/analysis/cell-cycle.html. It explains that our method is an implementation of the method used in [Macosko et al., 2015](https://www.ncbi.nlm.nih.gov/pubmed/26000488). You can see their paper for even more details.

Macosko et al. used the marker genes from [Whitfield et al., 2002](https://www.ncbi.nlm.nih.gov/pubmed/12058064). The Seurat genes used in their [cell cycle vignette](https://satijalab.org/seurat/v3.0/cell_cycle_vignette.html) are from [Tirosh et al., 2016](https://www.ncbi.nlm.nih.gov/pubmed/27124452), which are in turn a subset of the Whitfield genes. See the discussion in https://github.com/satijalab/seurat/issues/1139.

Since both `Seurat::CellCycleScoring()` and Macosko et al. use a marker gene approach to define the cell cycle, and they use similar marker genes as input, I think it makes sense that the results are similar.

But since you are interested in predicting the cell cycle phase, we'd be interested in getting your feedback on the [peco](https://github.com/jhsiao999/peco) method. How easy was it to use? How similar were the results to the other methods you have tried? Thanks!

**Comment by ateissan** [comment]
Thanks @jdblischak for your response. I am interested in trying new method to confirm my preliminary results. However, I am not sure that I can use peco. I am working with mouse single cell RNA-seq data. And I don't know how to train my samples. Can you help me to understand how to run the peco R package without a human dataset?

**Comment by jhsiao999** [comment]
Hi @ateissan, thanks for your interest in our software. We applied the method to human data from embryonic stem cells and benchmark against other methods for cell-cycle discrete classification (Seurat and Cyclone). The results showed that peco is able to pick out cell-cycle signals in single-cell RNA-seq data from embryonic stem cells and also to produce classification that is consistent with reporter-based classification using FUCCI. 
 
What are the methods that you have tried to perform cell-cycle phase classification?

In addition, I am working on an update of the software. The new feature will incorporate our method into the analysis workflow of SingleCellExperiment S4 objects. 

Joyce

**Comment by ateissan** [comment]
I tried the Seurat method (CellCycleScoring function). For peco, I don't have any trained data with mouse samples. Can I still use your method?

Aurélie

**Comment by jhsiao999** [comment]
@ateissan Yes. You can still apply the method. What cell type do you have? And how sparse is the data (i.e., fraction of zero entries in the count matrix)?
